Automattic/woocommerce-payments
# Add and config Jetpack IDC package

Fixes #3566


#### Changes proposed in this Pull Request

- Move WCPay SVG logo to a separate file to re-use it in the IDC configuration.
- Add `jetpack-identity-crisis` package and its configuration.
- Add JetpackIdcNotice to the `Overview` page. I add this because WooCommerce core (via its component woocommerce-admin) tries to hide admin notices (including Jetpack IDC banner) for React-powered setting pages _intentionally_ - [src](https://github.com/woocommerce/woocommerce-admin/blob/0b68c1d7c0c5025789071022efa129126da1ded5/src/Loader.php#L724-L739).
- Document explaining Safe Mode - 2b6c0-pb . It will be available here https://woocommerce.com/document/payments/faq/safe-mode/

#### Screenshots: 

Overview page:
![overview](https://user-images.githubusercontent.com/10045087/150748570-d4923a84-340a-43ac-bb1e-28251ee15f57.png)

Banner for non-admin users: 
![non-admin](https://user-images.githubusercontent.com/10045087/150748586-4c69a3ea-9735-46cb-8fdf-e6156f231214.png)

Banner for admins:
![admin](https://user-images.githubusercontent.com/10045087/150748634-a1af896a-4bc5-4308-8ab4-99269d4651af.png)

Orange button in the admin bar when choosing `Or decide later and stay in Safe mode`
![Markup on 2022-01-28 at 10:30:36](https://user-images.githubusercontent.com/10045087/151482571-634c68d8-e6f6-405d-b2b8-fa142aae91e4.png)


After taking actions:
![after](https://user-images.githubusercontent.com/10045087/150748657-6f18c5da-d943-43ed-98ca-992ccede4317.png)










#### Testing instructions





Mostly based on p9dueE-431-p2#comment-6978 and p9dueE-431-p2#comment-6957

1. Install `Jetpack Debug Helper` https://github.com/Automattic/jetpack-debug-helper
2. Checkout this PR. 
3. Run Jurassic / ngrok sites (as the IDC banner will NOT trigger with 
4. In wp-admin, click the `Jetpack Debug` menu item.
5. Enable the Identity Crisis Simulation Utility.
6. Navigate to wp-admin -> Jetpack Debug -> IDC Simulator
7. Enable the IDC simulation (you can enter a spoof url if you’d like, or just use the default spoof url, which is `https://example.org”.
8. Click “Send Remote Request” button in the IDC Simulator. OR ideally, we can trigger a REST request by visiting WCPay Overview/Transaction/Deposit pages. 
9. **Expectation: The IDC should be detected, and the UI should be displayed.**
10. If possible, install, build, and activate Jetpack in wp-content/plugins. 
11. **Expectation: WCPay’s IDC UI is displayed everywhere except the Jetpack dashboard, which displays Jetpack’s IDC UI.**
12. Try to choose an option and see the result. 

**Pre merge**

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply) - will do that later to avoid conflicts

**Post merge**
